### PR TITLE
mrd-2329_file-management-integration: supporting documents now using …

### PR DESCRIPTION
**Pull request by bills7281**
…file management api

### CODE REVIEW

**Review by DarrenW-HMPPS — CHANGES_REQUESTED** [review]

File: fake-document-management-api/mappings/document-management-download.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "request": {
+    "method": "GET",
+    "urlPattern": "/documents.*"
+  },
+  "response": {
+    "status": 201,
```
[the remark this answers is not in the document]
Comment (bills7281) [reply fake-document-management-api/mappings/document-management-download.json]: Corrected!

File: fake-document-management-api/mappings/document-management-delete.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "request": {
+    "method": "DELETE",
+    "urlPattern": "/documents.*"
```
[the remark this answers is not in the document]
Comment (bills7281) [reply fake-document-management-api/mappings/document-management-delete.json]: Corrected!

File: fake-document-management-api/mappings/document-management-download.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "request": {
+    "method": "GET",
+    "urlPattern": "/documents.*"
```
[the remark this answers is not in the document]
Comment (bills7281) [reply fake-document-management-api/mappings/document-management-download.json]: Corrected!

File: fake-document-management-api/mappings/document-management-download.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  },
+  "response": {
+    "status": 201,
+    "body": "SGVsbG8sIFdvcmxkIQ==",
+    "headers": {
+      "Content-Type": "application/octet-stream"
```
[the remark this answers is not in the document]
Comment (bills7281) [reply fake-document-management-api/mappings/document-management-download.json]: Corrected!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/client/DocumentManagementClient.kt [hunk 5]
```diff
@@ -31,6 +32,49 @@ class DocumentManagementClient(
...
+        )
+      }
+      .withRetry()
+  }
+
+  fun deleteFile(documentUuid: String?): Mono<Void> {
+    log.info(StringUtils.normalizeSpace("Deleting document with UUID $documentUuid"))
+    return webClient.delete()
+      .uri("/documents/$documentUuid")
+      .header("Service-Name", "Consider a recall")
```
[the remark this answers is not in the document]
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/client/DocumentManagementClient.kt:58]: From the Slack chat it looks like this is ok now.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/IntegrationTestBase.kt [hunk 6]
```diff
@@ -872,6 +875,36 @@ abstract class IntegrationTestBase {
     )
   }
 
+  protected fun documentManagementApiDownloadResponse(responseBody: String = "hello there!", delaySeconds: Long = 0) {
+    val responseBodyBytes = responseBody.toByteArray()
+
+    val documentManagementApiRequest = request()
+      .withMethod("GET")
+      .withPath("/documents.*")
```
[the remark this answers is not in the document]
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/IntegrationTestBase.kt]: Corrected!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/IntegrationTestBase.kt [hunk 7]
```diff
@@ -872,6 +875,36 @@ abstract class IntegrationTestBase {
...
+      response().withStatusCode(200)
+        .withBody(responseBodyBytes)
+        .withDelay(Delay.seconds(delaySeconds)),
+    )
+  }
+
+  protected fun documentManagementApiDeleteResponse(delaySeconds: Long = 0) {
+    val requestMatcher = request()
+      .withMethod("DELETE")
+      .withPath("/documents.*")
```
[the remark this answers is not in the document]
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/IntegrationTestBase.kt]: Corrected!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/IntegrationTestBase.kt [hunk 8]
```diff
@@ -872,6 +875,36 @@ abstract class IntegrationTestBase {
...
+    // Add any additional matching criteria here, such as headers
+
+    val response = response()
+      .withStatusCode(204)
+      .withBody("")
```
[the remark this answers is not in the document]
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/IntegrationTestBase.kt]: Removed

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/SupportingDocumentControllerTest.kt [hunk 9]
```diff
@@ -66,13 +66,13 @@ class SupportingDocumentControllerTest(
     val results = documentManagementApi.retrieveLogMessages(request())
 
     val result = recommendationSupportingDocumentRepository.findById((response.get("id") as Int).toLong())
-    assertThat(String(result.get().data)).isEqualTo("While I pondered, weak and weary")
     assertThat(result.get().documentUuid).isNotNull
   }
 
   @Test
   fun `retrieve supporting documents`() {
```
[the remark this answers is not in the document]
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/SupportingDocumentControllerTest.kt:72]: Yeh, good point. I didn't introduce this function in my PR, so would it be ok if I addressed it outside the scope if this work?

**Review by DarrenW-HMPPS — APPROVED** [review]

**Pull request merged**